### PR TITLE
Use @lumino/datagrid for the variable table view

### LINKED ISSUE #366: Fixed postion of header of table during scroll
noSteekyheader

### LINKED ISSUE #388: Switch to @lumino/datagrid for the table view
As mentioned in [this comment](https://github.com/jupyterlab/debugger/pull/369#issuecomment-594065552), we should consider switching to the Lumino data grid to display variables as a way to unify table components across JupyterLab.

### PR DESCRIPTION
**Pull request by KsavinN**
Fixes #388
Fixes #366 
Replaces #369
 
dataGrid

### CODE REVIEW

**Comment by jtpio** [comment]
Putting this here as reference. This is how the `csvviewer-extension` in JupyterLab handles the styling of the grid based on the theme:

https://github.com/jupyterlab/jupyterlab/blob/d8c66c009df2c04ceff454f88e25fe1650b58312/packages/csvviewer-extension/src/index.ts#L260-L302

**Comment by afshin** [comment]
cf. https://github.com/jupyterlab/debugger/issues/366

**Comment by jtpio** [comment]
Trying this PR locally, there seems to be a couple of items left to reach parity with the previous table viewer:

- being able to scroll the list of variables in the table
- opening variables as a main area widget on double click

**Comment by jtpio** [comment]
Rebased and updated the PR with:

- a couple of tweaks and cleanups
- CSS tweaks
- use the double click instead of single click to open variables 
- tweak the datagrid style colors so they match the style of the CSV viewer

datagrid

File: src/tokens.ts [hunk 1]
```diff
@@ -124,10 +124,11 @@ export interface IDebugger {
...
 
   /**
-   * Request details for a variable.
-   * @param variable The variable for which to request details.
+   * Request variables for a given variable reference.
+   *
+   * @param variablesReference The variable reference to request.
    */
-  getVariableDetails(
+  inspectVariable(
```
Comment (jtpio) [inline src/tokens.ts:131]: This is technically a breaking change, but it should be fine to make for `0.3.0` (still `0.x`) as it clarifies the interface.

**Comment by jtpio** [comment]
The last commits include a couple more cleanups.

This should now be ready (cc @afshin)

**Pull request merged**